src/server/cea_608_encoder/cea_608_utility.py

```diff
@@ -56,19 +59,68 @@ def bytes_to_byte_pairs(byte_list: list) -> list:
     return byte_pairs
 
 
-def create_byte_pairs(caption_string: str) -> list:
+def which_char_set(caption_char: str) -> str:
+    """Finds which character set the letter is in
+
+    :param caption_char:
+    :return: the name of the char set the caption_char is in
+    """
+    for char_set_name,list_of_characters in char_sets.items():
+        if caption_char in list_of_characters:
+            return char_set_name
+
+
+def which_channel(channel_toggle: int,char_set: str) -> hex:
+    """Provides the correct first byte to a letter depending on the channel toggle
+
+    :param channel_toggle, char_set:
+    :return: The first byte
+    """
+    if channel_toggle == 0:
+        if char_set == 'basic_na_set':
+            return 0x00
+        elif char_set == 'special_na_set':
+            return 0x11
+        elif char_set in ('extended_we_sm_set','extended_we_french_set'):
+            return 0x12
+        elif char_set in ('extended_we_port_set','extended_we_gd_set'):
+            return 0x13
+        #raise ValueError(f'The character set: {char_set} is not supported')
+    elif channel_toggle == 1:
+        if char_set == 'basic_na_set':
+            return 0x00
+        elif char_set == 'special_na_set':
+            return 0x19
+        elif char_set in ('extended_we_sm_set','extended_we_french_set'):
+            return 0x1a
+        elif char_set in ('extended_we_port_set','extended_we_gd_set'):
+            return 0x1b
+        #raise ValueError(f'The character set: {char_set} is not supported')
+    else:
+        raise ValueError(f'Channel toggle must be 0 or 1!')
+
+
+def create_byte_pair(caption_string: str, channel_toggle: int) -> list:
     """Generates a list of byte pairs given a caption string
 
-    :param caption_string:
+    :param caption_string, channel_toggle:
     :return: list of byte pairs
     """
     byte_list = []
     for letter in caption_string:
-        if letter in basic_north_american_char_set:
-            character_hex_value = basic_north_american_char_set[letter]
-            if check_parity(character_hex_value) == 0:
-                character_hex_value = add_parity_to_byte(character_hex_value)
-            byte_list.append(hex(character_hex_value))
+        char_set_name = which_char_set(letter)
+        first_byte = which_channel(channel_toggle,char_set_name)
+        if first_byte != 0x00:
```

Comment:
this line makes me uneasy because a reader of the code will be wondering why a first_byte cant be 0, and to find out they will have to dig through the code to see why 0x00 is marked only for characters part of the regular north American set. 

I can't think of a clever way around this at the moment. Let's remove the comments from the `which_channel` function and remove the print and hard coded caption at the bottom and ill approve and let you `squash and merge` in the merge menu, so all your commits become squashed into one.